lisa/variable.py

```diff
@@ -81,15 +81,16 @@ def load_from_file(
     is_secret: bool = False,
 ) -> None:
     if is_secret:
-        secret.add_secret(file_name)
+        secret.add_secret(file_name, secret.PATTERN_FILENAME)
 
     path = constants.RUNBOOK_PATH.joinpath(file_name)
-    if not path.exists:
-        raise FileNotFoundError(path)
 
     if path.suffix.lower() not in [".yaml", ".yml"]:
         raise LisaException("variable support only yaml and yml")
 
+    if not path.exists():
+        raise FileNotFoundError(f"cannot find variable file: {path}")
+
```

Comment:
```suggestion

```

Checking if a file exists before opening it is an anti-pattern that causes time-of-check-time-of-use problems.  `open()` will fail if the file doesn't exist; if you want to throw this particular exception, you can catch it with a `try` block below

Reply:
fixed

Reply:
It's fixed and pushed in a commit `remove anti-pattern path exists check before open`. It fixes some other instances.